fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -3349,6 +3349,33 @@ ACTOR Future<Void> processTransactionStateRequestPart(TransactionStateResolveCon
 
 } // anonymous namespace
 
+// Logs metrics related to workload burstiness on a much more frequent interval
+// than the typical five second ProxyMetrics trace.
+ACTOR Future<Void> logDetailedMetrics(ProxyCommitData const* commitData) {
```

Comment:
Is it not possible to use the same traceCounters function we use for regular metrics? Is it possible to have a counter belong to multiple CounterCollection's?

Reply:
Good point, I just looked into this a little. I think the main problem would be that the rate and roughness calculations for `ProxyMetrics` would be thrown off, because the shared `Counter` would now be getting reset on 100ms and 5s intervals (so essentially just looks like its getting reset every 100ms).

There's no infrastructure as far as I can tell to allow a `Counter` to support belonging to multiple `CounterCollections`. And I believe any change there would have to include changes to `Counter`, as each `Counter` keeps track of its own interval right now (somewhat indirectly, through a reset function being called).

Based on the above, the quick way to add these new metrics is the current state of the PR with the new logging loop and manually calculating the deltas, or by creating new `Counter`s that increment in the same place as the `Counter`s we want to track. I'm not sure which solution is better, they both are non-ideal.